### PR TITLE
[#92]Release Preparations v5.0

### PR DESCRIPTION
**Pull request by dwhite4**
- Reorganized.
- Added error handling.
- Reworked UI.
- Removed old/unneeded code.
- Removed unused "hbanim.py".
- Reworked export options.
- Refactored CAF to I_CAF.

### CODE REVIEW

File: io_export_cryblend/__init__.py [hunk 1]
```diff
@@ -488,39 +363,73 @@ def getMaterialPhysics():
...
-        empty_object["endframe"] = self.end_frame
+        active = bpy.context.active_object
 
-        if object_:
-            object_.select = True
-            bpy.context.scene.objects.active = object_
+        if (active.type == "MESH"):
+            already_exists = False
+            for object_ in bpy.data.objects:
+                if (object_.name == "{0}_{1}-proxy".format(active.name, getattr(self, "type_")) or
```
Comment (travnick) [inline io_export_cryblend/__init__.py]: I'd like to see it similar to this: 
```python
if is_proxy(object_.name):
...

def is_proxy(name):
    return object_.name == "{0}_{1}-proxy".format(active.name, getattr(self, "type_"))
                or object_.name.endswith("-proxy")
```

File: io_export_cryblend/__init__.py [hunk 2]
```diff
@@ -488,39 +363,73 @@ def getMaterialPhysics():
...
+        bound_box.name = "{0}_{1}-proxy".format(object_.name, getattr(self, "type_"))
+        bound_box.draw_type = "WIRE"
+        bound_box.dimensions = object_.dimensions
+        bound_box.location = object_.location
+        bound_box.rotation_euler = object_.rotation_euler
+
+        for group in object_.users_group:
+            bpy.ops.object.group_link(group=group.name)
+
+        proxy_material = bpy.data.materials.new("{0}_{1}-proxy__physProxyNone".format(object_.name, getattr(self, "type_")))
```
Comment (travnick) [inline io_export_cryblend/__init__.py]: sth like this looks better (line too long):

```python
name = "{0}_{1}-proxy__physProxyNone".format(object_.name, getattr(self, "type_"))
proxy_material = bpy.data.materials.new(name)
```

File: io_export_cryblend/__init__.py [hunk 3]
```diff
@@ -192,40 +178,30 @@ class AddCryExportNode(bpy.types.Operator):
...
+             "Static Geometry"),
+            ("cga", "CGA",
+             "Animated Geometry"),
+            ("chr", "CHR",
+             "Character"),
+            ("skin", "SKIN",
+             "Skinned Render Mesh"),
+            ("anm", "ANM",
+             "Geometry Animation"),
+            ("i_caf", "I_CAF",
```
Comment (travnick) [inline io_export_cryblend/__init__.py]: Why `i_caf` but not simple `caf`?

File: io_export_cryblend/export.py [hunk 4]
```diff
@@ -1198,22 +1214,53 @@ def write_to_file(config, doc, file_name, exe):
...
         file.close()
 
 
+def write_scripts(config, filepath):
+    if not config.make_chrparams and not config.make_cdf:
+        return
+
+    dae_path = utils.get_absolute_path_for_rc(filepath)
+    components = dae_path.split("\\")
+    dae_name = components[len(components)-1]
```
Comment (travnick) [inline io_export_cryblend/export.py]: in python it's `components[-1]`

File: io_export_cryblend/export.py [hunk 5]
```diff
@@ -1198,22 +1214,53 @@ def write_to_file(config, doc, file_name, exe):
...
+    output_path = dae_path[:-len(dae_name)]
```
Comment (travnick) [inline io_export_cryblend/export.py]: what about `os.path.dirname(dae_path)`?

File: io_export_cryblend/utils.py [hunk 6]
```diff
@@ -542,6 +545,14 @@ def is_export_node(nodename):
...
+    nodenames = []
+    for group in get_export_nodes():
+        nodenames.append(get_node_name(group.name))
+    unique_nodenames = set(nodenames)
+    if len(unique_nodenames) < len(nodenames):
+        return True
+
 def get_node_type(groupname):
     node_components = groupname.split(".")
     return node_components[len(node_components)-1]
```
Comment (travnick) [inline io_export_cryblend/utils.py]: `node_components[-1]`

File: io_export_cryblend/export.py (hunk 4 above)
Comment (dwhite4) [inline io_export_cryblend/export.py]: I never knew that.  Thanks for letting me know.  Pretty cool.  xD

File: io_export_cryblend/export.py (hunk 5 above)
Comment (dwhite4) [inline io_export_cryblend/export.py]: Good thinking, I will fix this.

**Pull request merged**